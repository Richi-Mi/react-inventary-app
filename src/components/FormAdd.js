import React, { useState } from 'react';
import PropTypes from 'prop-types';

const FormAdd = ({ setCells, cells }) => {
    const [inputNameValue, setInputNameValue]   = useState('');
    const [inputPriceValue, setinputPriceValue] = useState('');
    const [inputUnitValue, setinputUnitValue]   = useState('');

    const handleInputName = ( event ) => {
        setInputNameValue( event.target.value );
    }
    const handleInputPrice = ( event ) => {
        setinputPriceValue( event.target.value );
    }
    const handleInputUnit = ( event ) => {
        setinputUnitValue( event.target.value );
    }
    const addCategory = ( event ) => {
        event.preventDefault();

        if( inputNameValue !== '' && inputUnitValue !== '' && inputPriceValue !== '' ) {
            const data = {
                name: inputNameValue,
                units: inputUnitValue,
                price: inputPriceValue
            }
            let newCells = [...cells, data];
            setCells( newCells );
            
            setInputNameValue('');
            setinputPriceValue('');
            setinputUnitValue('');
        }
    } 

    return (
        <form className='form' onSubmit={ addCategory }>
            <h2> Agregar Categoria </h2>
            <section className='form_section' >
                <label> Escribe el Nombre   de la Categoria: </label>
                <input type='text' value={ inputNameValue } placeholder='Nombre' onChange={ handleInputName } id='txt_name'/>
            </section>
            <section className='form_section' >
                <label> Escribe el Precio   de la Categoria: </label>
                <input type='text' value={ inputPriceValue } placeholder='Precio' onChange={ handleInputPrice }/>
            </section>
            <section className='form_section' >
                <label> Escribe la Cantidad de la Categoria: </label>
                <input type='text' value={ inputUnitValue } placeholder='Cantidad' onChange={ handleInputUnit }/>
            </section>
            <section className='form_section_submit'>
                <input type={'submit'} value={'Agregar'} />
            </section>
        </form>
    )
}
FormAdd.propTypes = {
    setCells: PropTypes.func.isRequired,
    cells: PropTypes.array.isRequired
}

export default FormAdd