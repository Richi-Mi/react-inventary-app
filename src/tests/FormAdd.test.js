import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import FormAdd from '../components/FormAdd';

describe(' * Pruebas sobre el Componente <FormAdd /> ', () => {
    const setCells = jest.fn();
    const cells = [
    {
        name: 'Galletas', 
        units: 4, 
        price: 200
    },
    {
        name: 'Pan Danes', 
        units: 2, 
        price: 50
    },
    {
        name: 'Pastel', 
        units: 1, 
        price: 500
    }]
    let wrapper = shallow( <FormAdd setCells={ setCells } cells={ cells }/> );
    
    beforeEach( () => {
        jest.clearAllMocks()
        wrapper = shallow( <FormAdd setCells={ setCells } cells={ cells }/> );
    })
    
    test(' - Debe Mostrar el Componente Correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    test(' - NO Debe de Mandar informaciÃ³n el formulario si las condiciones no se cumplen', () => {
        const form = wrapper.find('.form');

        form.simulate('submit', { preventDefault: () => { } } );

        expect( setCells ).not.toHaveBeenCalled();
    })
})