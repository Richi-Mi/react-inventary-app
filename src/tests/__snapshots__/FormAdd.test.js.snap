// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas sobre el Componente <FormAdd />   - Debe Mostrar el Componente Correctamente 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h2>
     Agregar Categoria 
  </h2>
  <section
    className="form_section"
  >
    <label>
       Escribe el Nombre   de la Categoria: 
    </label>
    <input
      id="txt_name"
      onChange={[Function]}
      placeholder="Nombre"
      type="text"
      value=""
    />
  </section>
  <section
    className="form_section"
  >
    <label>
       Escribe el Precio   de la Categoria: 
    </label>
    <input
      onChange={[Function]}
      placeholder="Precio"
      type="text"
      value=""
    />
  </section>
  <section
    className="form_section"
  >
    <label>
       Escribe la Cantidad de la Categoria: 
    </label>
    <input
      onChange={[Function]}
      placeholder="Cantidad"
      type="text"
      value=""
    />
  </section>
  <section
    className="form_section_submit"
  >
    <input
      type="submit"
      value="Agregar"
    />
  </section>
</form>
`;
